Makefile

```diff
@@ -990,12 +995,12 @@ $(COCKROACH) $(COCKROACHOSS) $(COCKROACHSHORT) go-install:
 .PHONY: build buildoss buildshort
 build: ## Build the CockroachDB binary.
 buildoss: ## Build the CockroachDB binary without any CCL-licensed code.
-buildshort: ## Build the CockroachDB binary without the admin UI.
+buildshort: ## Build the CockroachDB binary without the admin UI, generated docs, or settings pages.
 build: $(COCKROACH)
 buildoss: $(COCKROACHOSS)
 buildshort: $(COCKROACHSHORT)
-build buildoss buildshort: $(if $(is-cross-compile),,$(DOCGEN_TARGETS))
-build buildshort: $(if $(is-cross-compile),,$(SETTINGS_DOC_PAGE))
+build buildoss: $(if $(is-cross-compile),,$(DOCGEN_TARGETS))
```

Comment:
I'd rather have the docgen/settings target remain generated without having to run `make build`.

The reason is that in a regular developer cycle, these need to be updated regularly. 

It's likely that a developer will 1) not run `make build` *at all* and at the same time 2) make ac hange that impacts the generated docs.

I don't know what a good solution is (yet). Have you ascertained how expensive these two targets are? If they are negligible,we could keep them in `buildshort`.

Reply:
Hm, I agree with you, though the numbers do leave me a bit sad.

```sh
# Building docgen/settings
$ time make cleanshort > /dev/null; time make buildshort > /dev/null; time make buildshort > /dev/null;
        5.83 real         3.99 user         5.86 sys
       32.77 real        38.59 user        12.41 sys
        5.66 real         7.78 user         6.17 sys

# Not building docgen/settings
$ time make cleanshort > /dev/null; time make buildshort > /dev/null; time make buildshort > /dev/null;
        6.45 real         4.19 user         6.24 sys
       28.03 real        31.62 user         7.84 sys
        2.21 real         3.22 user         4.15 sys
```

Reply:
I'll defer to you on final yay/nay here, removing the build-time ticking we had before already brings the incremental rebuild time down from 26s, so I'm much happier than I was before.

Reply:
Let's keep the docgen in then. In the common case (after a rebuild) we're looking at a difference of 3 seconds, which is OK. When we get some bazel in there, that problem will alleviate on its own.